--- a/sapi/micro/php_micro_fileinfo.c
+++ b/sapi/micro/php_micro_fileinfo.c
@@ -752,7 +752,12 @@ const char *micro_get_filename(void) {
     static char *self_filename = NULL;
     if (NULL == self_filename) {
         self_filename = malloc(PATH_MAX);
-        (void)realpath((const char *)getauxval(AT_EXECFN), self_filename);
+        ssize_t micro_n = readlink("/proc/self/exe", self_filename, PATH_MAX - 1);
+        if (micro_n > 0) {
+            self_filename[micro_n] = 0;
+        } else {
+            (void)realpath((const char *)getauxval(AT_EXECFN), self_filename);
+        }
     }
     return self_filename;
 }
